### ISSUE TITLE
Botkit changes type of my incoming message to "reconnect"

### ISSUE THREAD

**Issue by javasaur** [howdyai/botkit#1529]
Hi there,

I have a reconnect mechanism on client-side, which tries to reconnect for several times if server is down or if user has some network issues. Whenever I try to reconnect, a new websocket is opened and the same payload is sent:
{"type": "hello", ... (app relevant info)} 
If I debug the incoming messages, I can see that raw_message is the same I'm sending, but the actual type of message is changed to "reconnect", which breaks my app logic, as my controller is listening on "hello"

Is the type modification an expected behavior?

**Comment by benbrown** [comment]
Yes, this is the expected behavior. This allows Botkit to resume conversations that were already in progress but associated with the previous socket connection.

You should amend your code to also listen for reconnect events.  If you want to treat them the same way as hello or welcome_back events, that should be fine.
